### PR TITLE
dragonboard-820c.conf, qcom-apq8096.inc: added overwrite options ...

### PR DESCRIPTION
**Pull request by Skypodo**
Hi,
this patch adds the opportunity to overwrite values and change these in custom projects.

### CODE REVIEW

**Comment by ndechesne** [comment]
hey, thanks for your patch. Do you have a use case for such changes? i am not against making the change for some variables, but some of them really , i don't think should be changed.. they are here to describe the 'machine', and it won't change much... can you the variables that you really need to change? or provide more details why you need that?

also, a few cosmetic notes:
* please provide a 'better' commit log, about why you are making the change
* can you please send against master? if you need in a stable branch, let me know, and i will backport.

**Comment by Skypodo** [comment]
The reason for these changes is that I don't want to have to change the Variables within a config file but rather from _outside_. I changed all of them at once because I don't know, if I need to change other variables in future which could avoid additional pull requests.

Currently I would only need `SOC_FAMILY` from `qcom-apq8096.inc` and the following from `dragonboard-820c.conf`:
```
KERNEL_DEVICETREE
SERIAL_CONSOLE
QCOM_BOOTIMG_ROOTFS
```

> please provide a 'better' commit log

I'll keep that in mind for future commits.

> can you please send against master?

I'm using the _morty_ branch, that's why I made the pull request on it. Shall I make the pull request on the master anyway?

**Comment by ndechesne** [comment]
you still do no describe what you are doing here... from what you say it looks like you want to reuse db820c.conf file for a *different* board. which is not the right thing to do. if you have a custom/different board, you should make your own board .conf file instead. 

if i am wrong, please let me know.

if you have a db820c like board, and that the board is a commercial product, i am happy to merge a .conf file for that board.

**Comment by Skypodo** [comment]
I'm using the eval 820c Dragonboard but with a different kernel and, for example, with a different serial interface. To avoid an additional file which and to be able to set these via environment, I'd like to have these variables optional.

File: conf/machine/include/qcom-apq8096.inc [hunk 1]
```diff
@@ -24,8 +24,8 @@ PREFERRED_PROVIDER_virtual/xserver ?= "xserver-xorg"
 PREFERRED_PROVIDER_virtual/kernel ?= "linux-linaro-qcomlt-db820c"
 
 # Fastboot expects an ext4 image, which needs to be 4096 aligned
-IMAGE_FSTYPES_append = " ext4.gz"
-IMAGE_ROOTFS_ALIGNMENT = "4096"
+IMAGE_FSTYPES_append ?= " ext4.gz"
```
[the remark this answers is not in the document]
Comment (Skypodo) [reply conf/machine/include/qcom-apq8096.inc]: Yes, you're right. It should be `IMAGE_FSTYPES` I think.

**Comment by ndechesne** [comment]
ok, thanks for explaining your use case. using a different kernel is a valid concern. 

i would be ok to merge if your change adds ?= to KERNEL_* and SERIAL_CONSOLE variables. I don't think the other variables need it. can you update your PR, accordingly, and maybe add a bit more info in the commit msg?

If you can't send against master, it's fine. i will take care of applying the patch against master, once it's merged in this branch.

**Comment by ndechesne** [comment]
QCOM_BOOTIMG_ROOTFS is already set with ?= since I was expecting users to change it.
SOC_FAMILY should never be changed. If you need to change SOC_FAMILY, you definitely need your own machine .conf file, it's no longer a DB820c at all.. 

I am ok to change variables that can help you run a different kernel, easily.

**Comment by Skypodo** [comment]
I added my changes to the branch. That should be it for now.

**Comment by ndechesne** [comment]
thanks for your time! I have applied your change now. It have applied to master, sumo, rocko and morty. Also, i made some changes before committing:
* i squashed both patches into 1
* i extended your change to all machines, not just 820c.

see  dfe5d14, 16032f6 and 955ce26

**Pull request closed without merging**